Feature/horizontal list

XDRO opened the pull request:
## Describe your changes
Create Horizontal List component 
## Issue ticket number and link
EES-14
https://tripleten-apiary.atlassian.net/browse/EES-14
## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [ ] I have made any necessary updates to Storybook to accompany these changes
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

Commit:
initial commit

Commit:
updated package-lock

RPinkha committed:
feat: set-up fonts in default.css

RPinkha committed:
feat: added all the colors to the tailwind config

RPinkha committed:
feat: added fonts in the tailwind.config file

RPinkha committed:
feat: added fontSizes in the tailwind.config file

raymondafan committed:
checkbox/textareainput

RPinkha committed:
fix: removed styling that is available from default

raymondafan committed:
Add components to repository

raymondafan committed:
feat(checkbox-input): update checkbox styles in checkbox-textarea

RPinkha committed:
fix: removed unneeded import and deprecated  options

RPinkha committed:
fix: combined the two content arrays into one

tomershvueli committed:
Merge pull request #6 from tomershvueli/feature/update-tailwind-theme

Feature/update tailwind theme

raymondafan committed:
Merge branch 'main' of https://github.com/tomershvueli/expert-event-solutions into feature/checkbox-textarea

raymondafan committed:
feat: completed merge

Commit:
Revert "feat: completed merge"

This reverts commit 73d2b5e492ee0ce0aa200998e78c69f2588b454f.

Commit:
Revert "Merge branch 'main' of https://github.com/tomershvueli/expert-event-solutions into feature/checkbox-textarea"

This reverts commit b964655ab6e97a145a3bc960f46293a328cd052b, reversing
changes made to 519ca4c181ab6bec2b17d23809b5e119e3456580.

Commit:
Reapply "Merge branch 'main' of https://github.com/tomershvueli/expert-event-solutions into feature/checkbox-textarea"

This reverts commit c1dbbde2f87b3a4e280de3a5e850e132f9b12d37.

Commit:
remove checkbox and textarea from this branch

Commit:
begin working on horizontal list

Commit:
get a generic structure together for HorizontalList and HorizontalListItem

Commit:
create custom class names for each list item, begin working on one list

Commit:
add styles

Commit:
feature: update styles

Commit:
push after working on responsive design and creating story

Commit:
update styles to create responsive layout for horizontal list

Commit:
create staffing services array and style it for wider screen widths

Commit:
create responsive design staffing services list for mobile widths, and  create new array to be used for mobile width

Commit:
create new stories file for staffing services list

Commit:
track new staffing services stories file

Commit:
remove unused key prop

Commit:
Merge branch 'main' into feature/horizontal-image-list accept incoming tailwind.config.js

raymondafan commented on src/components/App/App.tsx:
```diff
@@ -0,0 +1,230 @@
+
+  return (
+    <div>
+      {/* Home Page List */}
```
Hey Joshua so I've observed that the layout breaks at 1130px, with text overflowing from the container. This issue might stem from our Tailwind CSS breakpoints. We should should probably update the breakpoints in tailwind.config.js and revise the styles in the HorizontalList component to prevent overflow. After making these tweaks and testing the sizes I'm sure we can merge the changes to main as soon as possibe.
<img width="1440" alt="Screenshot 2024-09-18 at 10 50 59 PM" src="https://github.com/user-attachments/assets/a4557d56-2b7a-4abf-b880-58091ec1b237">

raymondafan commented on src/components/App/App.tsx:
```diff
@@ -0,0 +1,230 @@
+          />
+        ))}
+      </HorizontalList>
+      {/* Staffing Services List*/}
```
The layout also breaks at 1357px for the other list. We might need a final adjustment to the Tailwind CSS media queries in tailwind.config.js to address this issue. <img width="1440" alt="Screenshot 2024-09-18 at 11 22 01 PM" src="https://github.com/user-attachments/assets/8ed521f6-f14b-43d0-9966-7fee1287516b">

raymondafan reviewed (commented):
The code looks great overall and is structured well. There are just a few minor adjustments needed to address layout issues at specific breakpoints. Specifically, we need to review and potentially update the Tailwind CSS breakpoints to ensure proper responsiveness. Once these adjustments are made and tested, the code should be ready for final approval and merging into the main branch.

Commit:
add image component inside of HorizontalListItem; inside of app build an array, and map over it; ensure that src and alt attributes are present when mapping items

Commit:
get a layout for the desktop version for image horizontal list

michaelabudde reviewed (commented):
Hi there, looks good, two things that need fixing. The Merriweather font was not being applied. I played around with it and found that the issue was using font-merriweather rather than the name in tailwind which is font-title. Once I made those changes the font was applied.
The second thing I think Raymond already touched on but the responsive design just needs some fixing on intermediate screens. 
<img width="1379" alt="Screenshot 2024-09-19 at 11 03 15 AM" src="https://github.com/user-attachments/assets/6ded3c74-f6ad-4212-8acd-e98c74ac2ac8">
<img width="1379" alt="Screenshot 2024-09-19 at 11 02 36 AM" src="https://github.com/user-attachments/assets/ca3bee7f-934e-4af4-a49b-dec234ccf2de">
<img width="1379" alt="Screenshot 2024-09-19 at 11 02 28 AM" src="https://github.com/user-attachments/assets/ec104ffd-3805-426d-8fd9-0bc89dc24517">

Commit:
add more styling to image horizontal list component

Commit:
remove unused imports; create isMobile state tracker; fix fonts; conditionally render each horizontal image component

Commit:
begin creating stories file for horizontal image list

Commit:
refactor entire horizontal list component to create cleaner file structure with greater ease of reuse

Commit:
remove app import from main

Commit:
create a more responsive layout for HorizontalListImage component; refactor stories to import and use HorizontalListImage, without redefining it

Commit:
update stories component to use imported HorizontalListStaffing

Commit:
add styles for responsiveness; add flex-wrap;

Commit:
refactor horizontal list home stories

Commit:
begin making horizontal list home page more responsive

Commit:
refactor HorizontalListHome.tsx file; create more responsive layout

Commit:
clean up main.tsx file

XDRO commented:
I have addressed concerns and refactored the code, for greater reuse-ability and cleaner code. Easily comprehended and digested. I have also realigned this branch to match the one of `feature/horizontal-image-list` which refactored the file structure and use of this code, and cleaned up the main.tsx file. @michaelabudde @raymondafan @RPinkha @tomershvueli

Commit:
add justify center to horizontal home list syles

raymondafan commented:
Looks good, ready for final review!

raymondafan reviewed (approved):
looks good, ready for final review!

RPinkha committed:
fix: deleted stories folder

RPinkha committed:
fix: fixed polution from main

RPinkha committed:
fix: deleted checkbox images

tomershvueli commented on src/components/HorizontalList/HorizontalListHome/HorizontalListHome.tsx:
```diff
@@ -0,0 +1,71 @@
+            lg:px-[12px] lg:gap-[12px] 
+            md:px-[12px] md:gap-[12px]
+            sm:px-[12px] sm:gap-[12px]"
+            //
```
nit: unnecessary line

tomershvueli commented on src/components/HorizontalList/HorizontalListHome/HorizontalListHome.tsx:
```diff
@@ -0,0 +1,71 @@
+import { HorizontalList } from "../HorizontalList";
+export function HorizontalListHome() {
+  const HomePageItems = [
```
The items to render should be passed in to this component as a prop, not hardcoded in this file.

tomershvueli commented on src/components/HorizontalList/HorizontalListImage/HorizontalListImage.tsx:
```diff
@@ -0,0 +1,119 @@
+
+  const isMobile = windowWidth <= 850;
+
+  const aboutItems: AboutItem[] = [
```
This array should be passed in as a prop.

tomershvueli commented on src/components/HorizontalList/HorizontalListImage/HorizontalListImage.tsx:
```diff
@@ -0,0 +1,119 @@
+          unorderedListClassName="flex gap-[86px] pt-[10px] justify-center"
+        >
+          {aboutItems.map((item, index) =>
+            item.src && item.alt ? ( // Ensure src and alt exist
```
Having an `alt` on images is best practice, but if we don't have an `alt` we shouldn't not render the image.

tomershvueli commented on src/components/HorizontalList/HorizontalListImage/HorizontalListImage.tsx:
```diff
@@ -0,0 +1,119 @@
+          >
+            {aboutItems.map(
+              (item, index) =>
+                item.src && item.alt ? ( // Ensure src and alt exist
```
See comment above about `alt`.

tomershvueli commented on src/components/HorizontalList/HorizontalListHome/HorizontalListHome.tsx:
```diff
@@ -0,0 +1,71 @@
+import { HorizontalList } from "../HorizontalList";
+export function HorizontalListHome() {
```
Let's make this component more generic by not calling it `HorizontalListHome`, but rather something like plain `HorizontalList`.

tomershvueli commented on src/components/HorizontalList/HorizontalListStaffing/HorizontalListStaffing.tsx:
```diff
@@ -0,0 +1,140 @@
+import { HorizontalList } from "../HorizontalList";
+import { useEffect, useState } from "react";
+export function HorizontalListStaffing() {
```
Let's make this component more generic by not calling it `HorizontalListStaffing`, but rather something like `HorizontalListBackground`.

tomershvueli commented on src/components/HorizontalList/HorizontalListStaffing/HorizontalListStaffing.tsx:
```diff
@@ -0,0 +1,140 @@
+    return () => window.removeEventListener("resize", handleResize);
+  }, []);
+
+  const staffingItems =
```
This array should be a prop passed in.

RPinkha committed:
feat: merged main

RPinkha committed:
fix: removed comment
